photon-ml/src/main/scala/com/linkedin/photon/ml/avro/AvroUtils.scala

```diff
@@ -86,28 +97,89 @@ object AvroUtils {
     }
   }
 
-  protected[avro] def getNameAndTermFromAvroRecord(record: GenericRecord): NameAndTerm = {
+  /**
+   * Read the nameAndTerm of type [[NameAndTerm]] from Avro record of type [[GenericRecord]]
+   * @param record the input Avro record
+   * @return The nameAndTerm parsed from the Avro record
+   */
+  protected[avro] def readNameAndTermFromGenericRecord(record: GenericRecord): NameAndTerm = {
     val name = Utils.getStringAvro(record, AvroFieldNames.NAME, isNullOK = false)
     val term = Utils.getStringAvro(record, AvroFieldNames.TERM, isNullOK = false)
     NameAndTerm(name, term)
   }
 
-  protected[avro] def convertCoefficientsToBayesianLinearModelAvroRecord(
+  /**
+   * Parse a set of nameAndTerm of type [[NameAndTerm]] from a RDD of Avro record of type [[GenericRecord]] with the
+   * user specified feature section keys
+   * @param genericRecords the input Avro records
+   * @param featureSectionKey the user specified feature section keys
+   * @return a set of nameAndTerms parsed from the input Avro records
+   */
+  protected[avro] def readNameAndTermSetFromGenericRecords(
+    genericRecords: RDD[GenericRecord],
+    featureSectionKey: String): Set[NameAndTerm] = {
+
+    genericRecords.flatMap(_.get(featureSectionKey) match { case recordList: JList[_] =>
+      val nnz = recordList.size
+      val featureNameAndTermBuf = new mutable.ArrayBuffer[NameAndTerm](nnz)
+      val iterator = recordList.iterator
+      while (iterator.hasNext) {
```

Comment:
This is a good suggestion. The overhead comes with JavaConverters will have minimal impact, because this it is not the bottleneck of this function.